src/app_config.rs

```diff
@@ -0,0 +1,52 @@
+use serde::Deserialize;
+use std::{collections::HashMap, fs, path::PathBuf};
+
+pub(crate) const CONFIG_FILE_NAME: &str = "cargo-generate";
+
+#[derive(Deserialize)]
+pub(crate) struct AppConfig {
+    pub favorites: HashMap<String, FavoriteConfig>,
+}
+
+#[derive(Deserialize)]
+pub(crate) struct FavoriteConfig {
+    pub description: Option<String>,
+    pub git: Option<String>,
+    pub branch: Option<String>,
+}
+
+pub(crate) fn edit_config(path: Option<PathBuf>) -> anyhow::Result<()> {
+    if let Some(path) = path {
+        println!(
+            r#"Attempting to edit config at "{}""#,
+            path.as_path().display().to_string()
+        );
+        if !path.exists() {
+            std::fs::File::create(&path)?;
+        }
+        open::that(path)?;
+        Ok(())
+    } else {
+        anyhow::bail!("No configuration file specified")
+    }
+}
+
+pub(crate) fn read_config(path: Option<PathBuf>) -> anyhow::Result<Option<AppConfig>> {
+    match path {
+        Some(path) if path.exists() => {
+            let config_content = fs::read_to_string(&path)?;
+            let config_content = config_content.trim();
```

Comment:
those calls are chainable like `fs::read_to_string(&path)?.trim();`

Reply:
Nope - doing so makes the compiler complain, as `trim()` only returns a reference and thus we have issues with a temporary variable 😢 
Getting rid of the `trim()` makes the toml parser (serde) complain on an empty config file 😭

Options?

Reply:
Fixed - but not where suggested as that would introduce a temporary variable problem